lib/fizzy/instantiate.hpp

```diff
@@ -28,7 +28,18 @@ struct ExternalFunction
     FuncType type;
 };
 
-using table_elements = std::vector<std::optional<ExternalFunction>>;
+// Table element, which references a function in any instance.
+struct TableElement
+{
+    Instance* instance;
+    FuncIdx func_idx;
+    // This pointer is empty most of the time and is used only to keep instance alive in one edge
+    // case, when start function traps, but instantiate has already modified some elements of a
+    // shared (imported) table.
+    std::shared_ptr<Instance> shared_instance;
+};
+
+using table_elements = std::vector<std::optional<TableElement>>;
```

Comment:
Now having `TableElement`, you can check `element.instance` for empty and skip `std::optional`.

Reply:
Yeah, goo idea.